wlauto/workloads/googlephotos/__init__.py

```diff
@@ -0,0 +1,107 @@
+
+__version__ = '0.1.1'
+
+
+class Googlephotos(AndroidUiAutoBenchmark):
+
+    name = 'googlephotos'
+    package = 'com.google.android.apps.photos'
+    activity = 'com.google.android.apps.photos.home.HomeActivity'
+    view = [package + '/com.google.android.apps.consumerphotoeditor.fragments.ConsumerPhotoEditorActivity',
+            package + '/com.google.android.apps.photos.home.HomeActivity',
+            package + '/com.google.android.apps.photos.localmedia.ui.LocalPhotosActivity',
+            package + '/com.google.android.apps.photos.onboarding.AccountPickerActivity',
+            package + '/com.google.android.apps.photos.onboarding.IntroActivity']
+    description = """
+    A workload to perform standard productivity tasks with Google Photos.  The workload carries out
+    various tasks, such as browsing images, performing zooms, post-processing and saving a selected
+    image to file.
+
+    Test description:
+    1. Four images are copied to the device
+    2. The application is started in offline access mode
+    3. Gestures are performed to swipe between images and pinch zoom in and out of the selected
+       image
+    4. The Colour of a selected image is edited by selecting the colour menu, incrementing the
+       colour, resetting the colour and decrementing the colour using the seek bar.
+    5. A Crop test is performed on a selected image.  UiAutomator does not allow the selection of
+       the crop markers so the image is tilted positively, reset and then negatively to get a
+       similar cropping effect.
+    6. A Rotate test is performed on a selected image, rotating anticlockwise 90 degrees, 180
+       degrees and 270 degrees.
+    """
+
+    parameters = [
+        Parameter('dumpsys_enabled', kind=bool, default=True,
+                  description="""
+                  If ``True`` turns on the action logger which outputs
+                  timestamps to logcat for actions recorded in the workload.
+                  """),
+        Parameter('test_images', kind=list_or_string, mandatory=True,
+                  description="""
+                  A list of four jpeg files to be pushed to the device.
+                  """),
+
+    def validate(self):
+        super(Googlephotos, self).validate()
+        self.uiauto_params['package'] = self.package
+        self.uiauto_params['output_dir'] = self.device.working_directory
+        self.uiauto_params['dumpsys_enabled'] = self.dumpsys_enabled
+
+        if len(self.test_images) != 4:
+            raise ValidationError("This workload requires four test images")
+
+        for image in self.test_images:
+            if not image.endswith(('jpg', 'jpeg')):
+                raise ValidationError('{} must be a jpeg file'.format(image))
+
+    def initialize(self, context):
+        super(Googlephotos, self).initialize(context)
+
+        for image in self.test_images:
+            image_path = context.resolver.get(wlauto.common.resources.File(self, image))
```

Comment:
Would also make sense to allow users to specify the absolute path to images. Check if the path is a file, if not then try the resource resolvers?